ops/pebble.py

```diff
@@ -3211,6 +3310,49 @@ def get_notices(
         resp = self._request('GET', '/v1/notices', query)
         return [Notice.from_dict(info) for info in resp['result']]
 
+    def get_identities(self) -> Dict[str, Identity]:
+        """Get all identities in Pebble.
+
+        Returns:
+            A dict mapping identity names to :class:`Identity` objects.
+
+        Raises:
+            :class:`ops.pebble.IdentityError`: If the server returns null for any identity.
+        """
+        resp = self._request('GET', '/v1/identities')
+        result = resp['result']
+
+        identities: Dict[str, Identity] = {}
+        for name, identity in result.items():
+            if identity is None:
+                raise IdentityError(f'server returned null identity {name!r}')
+            identities[name] = Identity.from_dict(identity)
+        return identities
+
+    def replace_identities(
+        self, identities: Dict[str, Union[IdentityDict, Identity, None]]
+    ) -> None:
+        """Replace the named identities in Pebble with the given ones.
+
+        Add those identities if they don't exist, or remove them if the map value is None.
+
+        Args:
+            identities: A dict mapping identity names to dicts or :class:`Identity` objects.
+        """
+        identities_dict: Dict[str, Optional[IdentityDict]] = {}
+        for name, identity in identities.items():
+            if identity is None or isinstance(identity, dict):
+                identities_dict[name] = identity
+            elif isinstance(identity, Identity):
+                identities_dict[name] = identity.to_dict()
+            else:
+                raise TypeError(
+                    f'identity must be a dict or pebble.Identity, not {type(identity).__name__}'
+                )
```

Comment:
I'd probably simplify by inverting the if/else logic and dropping the explicit TypeError for simplicity (we rely on static type checking a lot):

```suggestion
        identities_dict = {
            name: identity.to_dict() if isinstance(identity, Identity) else identity
            for name, identity in identities.items()
        }
```

Reply:
This was to follow the same logic/coding style in `add_layer`, but I do agree that the suggested code looks much cleaner.